CppSamples/ChilliConnectCppDemo/HttpSystem.cpp

```diff
@@ -0,0 +1,153 @@
+#include "HttpSystem.h"
+
+#include <iostream>
+#include <string>
+#include <Windows.h>
+#include <winhttp.h>
+#include <codecvt>
+#include <locale>
+
+HttpSystem::HttpSystem()
+{
+}
+
+HttpSystem::~HttpSystem()
+{
+}
+
+wstring
+HttpSystem::CreateHeaderString(const map<string,string>& headers)
+{
+	string headerBlob = "Content-Type: application/json\r\n";
+
+	for (auto it = headers.begin(); it != headers.end(); ++it)
+	{
+		headerBlob += it->first;
+		headerBlob += ": ";
+		headerBlob += it->second;
+		headerBlob += '\r';
+		headerBlob += '\n';
+	}
+
+	wstring headerBlobWide(headerBlob.length() - 2, L' ');
```

Comment:
This was also from ChilliSource..I thought it was also removing the trailing \r\n on the last header so just decided to leave it.